sdk/search/azure-search-documents/src/main/java/com/azure/search/annotation/SearchableFieldProperty.java

```diff
@@ -0,0 +1,79 @@
+package com.azure.search.annotation;
+
+import com.azure.search.documents.models.AnalyzerName;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * The annotation is to indicate whether the field is a searchable field. The boolean field of isSearchable
+ * defaults to true if use the annotation.
+ */
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SearchableFieldProperty {
+    /**
+     * Optional arguments defines whether the field is a key field or not.
+     *
+     * @return True if it is the key of SearchField, and false by default for non-key field.
+     */
+    boolean isKey() default false;
+
+    /**
+     * Optional arguments defines whether the field is hidden or not.
+     *
+     * @return True if it is not retrievable, and false by default for retrievable field.
+     */
+    boolean isHidden() default false;
+
+    /**
+     * Optional arguments defines whether the field is facetable or not.
+     *
+     * @return True if it is facetable, and false by default for non-facetable field.
+     */
+    boolean isFacetable() default false;
```

Comment:
Shouldn't this be `true` to match the service documentation?

Reply:
No. We pass false by default. All values default to their defaults (bool false, int 0, etc.).

Reply:
Same for similar questions elsewhere.